src/TLuaInterpreter.cpp

```diff
@@ -309,6 +309,11 @@ int TLuaInterpreter::raiseEvent( lua_State * L )
             pE.mArgumentList.append( QString(lua_tostring( L, i )) );
             pE.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
         }
+        else if( lua_isboolean( L, i ) )
+        {
+            pE.mArgumentList.append( QString::number(lua_toboolean( L, i )) );
```

Comment:
My take on that would be that while the strings "true" and "false" are unlikely to arise very often as arguments, they're still *possible* arguments, and you shouldn't assume that they're meant as booleans. Since TLuaInterpreter::callEventHandler() determines how to handle the arguments based solely on the types listed in mArgumentTypeList, you should probably always be looking at both mArgumentList and mArgumentTypeList when debugging a TEvent, anyway. A design that encourages inferring the type from the string value doesn't seem like a good idea. Maybe that's overly paternalistic of me, I dunno.

In any case, we're talking about tiny differences in efficiency in what will surely be a rare debugging scenario, anyway, and I'd be quite surprised to hear of a case where the choice actually made a noticeable difference one way or the other in practice. My preference is to leave it as written, but I see it as essentially a stylistic choice, and I'm happy to defer to the opinion of the community, if anyone cares enough to weigh in.